### PR TITLE
test(SSR e2e): case when a call to Backend timeouts

### PR DESCRIPTION
**Pull request by Platonn**
Added a test case: when a call to Backend timeouts

btw. changes:
- replaced `callback` with `responseInterceptor` in ProxyUtils

fixes https://jira.tools.sap/browse/CXSPA-8386

### CODE REVIEW

File: projects/ssr-tests/src/ssr-testing.spec.ts [hunk 1]
```diff
@@ -19,11 +19,8 @@ describe('SSR E2E', () => {
 
   describe('With SSR error handling', () => {
     describe('Common behavior', () => {
-      beforeEach(async () => {
-        await SsrUtils.startSsrServer();
-      });
-
       it('should receive success response with request', async () => {
+        await SsrUtils.startSsrServer();
```
Comment (pawelfras) [inline projects/ssr-tests/src/ssr-testing.spec.ts]: `query:` can you help me understand what's the reason of server init from `beforeEach` to each test case?

File: projects/ssr-tests/src/ssr-testing.spec.ts [hunk 2]
```diff
@@ -65,19 +65,64 @@ describe('SSR E2E', () => {
       });
 
       it.skip('should receive response with status 500 if HTTP error occurred when calling other than cms/pages API URL', async () => {
```
Comment (pawelfras) [inline projects/ssr-tests/src/ssr-testing.spec.ts]: `minor:` is this `skip` intentional here? (I'm aware it's not part of the PR itself)

File: projects/ssr-tests/src/ssr-testing.spec.ts (hunk 1 above)
Comment (Platonn) [reply projects/ssr-tests/src/ssr-testing.spec.ts]: I've added a new test under the `describe('With SSR error handling')`. In this specific test I needed to call `SsrUtils.startSsrServer(); with specific config , i.e. `SsrUtils.startSsrServer({timeout: 20_000});`

Then I couldn't have startSsrServer both in the overarching `beforeEach` and in this specific test.

File: projects/ssr-tests/src/ssr-testing.spec.ts (hunk 2 above)
Comment (Platonn) [reply projects/ssr-tests/src/ssr-testing.spec.ts]: It's deliberate. We've marked this test as flaky in last days.
See https://jira.tools.sap/browse/CXSPA-8570

File: projects/ssr-tests/src/ssr-testing.spec.ts (hunk 1 above)
Comment (pawelfras) [reply projects/ssr-tests/src/ssr-testing.spec.ts]: `nitpick:` Might be worth considering introducing a new section `describe('Without server timeout')` inside `describe('With SSR error handling')` and throwing all tests there including beforeEach, for which timeout is not required. You won't have to remember to start the server for these types of new tests.

**Review by pawelfras — APPROVED** [review]
LGTM. QA done.

**Review by pawelfras — DISMISSED** [review]
LGTM. QA done.
One nitpick: https://github.com/SAP/spartacus/pull/19307#discussion_r1782951174

**Review by pawelfras — APPROVED** [review]

**Comment by Platonn** [comment]
`SSR Tests` passed after running full pipeline on this branch: https://github.com/SAP/spartacus/actions/runs/11161307908/job/31023594584#step:7:53
